sims/benchmark-simuPOP.py

```diff
@@ -0,0 +1,330 @@
+                        for p in (0,1)]
+node_ids = {x:j for x, j in zip(haploid_labels, init_ts.samples())}
+rc = RecombCollector(ts=init_ts, node_ids=node_ids,
+                     locus_position=locus_position)
+end_prep = time.time()
+time_dict['prepping'] = end_prep - start
+
+pop.evolve(
+    initOps=[
+        sim.InitSex(),
+    ]+init_geno,
+    preOps=[
+        sim.PyOperator(lambda pop: rc.increment_time() or True),
+        sim.SNPMutator(u=args.sel_mut_rate, v=args.sel_mut_rate),
+        # so that selector returns, for f_i fitness values, \prod_i f_i
+        sim.PyMlSelector(GammaDistributedFitness(shape=args.gamma_shape,
+                                                 scale=args.gamma_scale),
+                         mode=sim.MULTIPLICATIVE),
+    ],
+    matingScheme=sim.RandomMating(
+        ops=[
+            id_tagger,
+            sim.Recombinator(intensity=args.recomb_rate,
+                output=rc.collect_recombs,
+                infoFields="ind_id"),
+        ] ),
+    postOps=[
+        sim.Stat(numOfSegSites=sim.ALL_AVAIL, step=50),
+        # sim.PyEval(r"'Gen: %2d #seg sites: %d\n' % (gen, numOfSegSites)", step=50),
+        sim.PyOperator(lambda pop: rc.simplify(pop.indInfo("ind_id")) or True,
+                       step=args.simplify_interval),
+    ],
+    gen = args.generations
+)
+end_sim = time.time()
+time_dict['simulating'] = end_sim - end_prep
+
+logfile.write("Done simulating!\n")
+logfile.write(time.strftime('%X %x %Z')+"\n")
+logfile.write("----------\n")
+logfile.flush()
+
+locations = [pop.subPopIndPair(x)[0] for x in range(pop.popSize())]
+rc.add_locations(pop.indInfo("ind_id"), locations)
+
+logfile.write("Collecting samples:\n")
+logfile.write("  " + str(args.nsamples) + " of them")
+logfile.write("  " + "ids:" + str(pop.indInfo("ind_id")))
+logfile.write("  " + "locations:" + str(locations))
+
+diploid_samples = random.sample(pop.indInfo("ind_id"), args.nsamples)
+rc.simplify(diploid_samples)
+
+del pop
+
+logfile.write("Samples:\n")
+logfile.write(str(rc.diploid_samples)+"\n")
+logfile.write("----------\n")
+logfile.flush()
+
+ts = rc.args.tree_sequence()
+del rc
+
+logfile.write("Loaded into tree sequence!\n")
+logfile.write(time.strftime('%X %x %Z')+"\n")
+logfile.write("----------\n")
+logfile.flush()
+
+if args.treefile is not None:
+    ts.dump(args.treefile)
+
+logfile.write("Writing out samples.\n")
+logfile.write(time.strftime('%X %x %Z')+"\n")
+logfile.write("----------\n")
+logfile.flush()
+
+mut_seed=args.seed
+logfile.write(time.strftime('%X %x %Z')+"\n")
+logfile.write("Generating mutations with seed "+str(mut_seed)+"\n")
+logfile.flush()
+
+rng = msprime.RandomGenerator(mut_seed)
+nodes = msprime.NodeTable()
+edgesets = msprime.EdgesetTable()
+sites = msprime.SiteTable()
+ts.dump_tables(nodes=nodes, edgesets=edgesets)
+if args.record_neutral:
+    mutated_ts = msprime.load_tables(nodes=nodes, edgesets=edgesets,
+                                     sites=sites)
```

Comment:
Can't work, currently. Need

- way for recombcollector to send mutation info to argrecorder (todo in ftprime)
- more setup above to record netural mutations

Reply:
I agree it'd be a better point of comparison to just have simuPOP do everything.